src/test/objectstore/allocsim/ops_parser.h

```diff
@@ -0,0 +1,77 @@
+#ifndef OPS_PARSER_H
+#define OPS_PARSER_H
+#include <cstdint>
+#include <vector>
+#include <string>
+#include <set>
+#include "include/rados/librados.hpp"
+#include <functional>
+
+
+// compare shared_ptr<string>
+struct StringPtrCompare
+{
+  int operator()(
+    const std::shared_ptr<std::string>& lhs, 
+    const std::shared_ptr<std::string>& rhs) const {
+    if (lhs && rhs) {
+        // Compare the content of the strings
+        return *lhs < *rhs;
+    }
+    return lhs < rhs;
```

Comment:
comparing shared pointers (the pointers themselves) was deprecated in C++20, and will be removed in c++26